subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/daemon/AbstractDaemonCompiler.java

```diff
@@ -47,7 +51,7 @@ public AbstractDaemonCompiler(File daemonWorkingDir, Compiler<T> delegate, Worke
     public WorkResult execute(T spec) {
         DaemonForkOptions daemonForkOptions = toDaemonOptions(spec);
         Worker<WorkerCompileSpec<?>> worker = workerFactory.getWorker(CompilerDaemonServer.class, daemonWorkingDir, daemonForkOptions);
-        DefaultWorkResult result = worker.execute(new WorkerCompileSpec<T>(delegate, spec));
+        DefaultWorkResult result = worker.execute(new WorkerCompileSpec<T>(delegate, spec, OperationIdentifierRegistry.getCurrentOperationId()));
```

Comment:
We want the worker stuff to work in terms of build operations, not progress events. It should be the business of the build operation stuff to bridge across whatever it needs into the progress events.

So, this would mean shipping around a _build operation id_ rather than a _progress operation id_. It would also mean adding a way to tell the `BuildOperationProcessor` what the parent build operation id is rather than setting the current progress operation id.

However, I would step back and consider whether it is the best approach to send the parent build operation id across to the worker process and then sending it back in every event. It might make more sense to simply attach the parent operation id to events whose build operation id is null received from the worker process.

Whatever the solution, we should do it at a lower level than `AbstractDaemonCompiler`, so that it works for _all_ worker processed, and so that it is transparent to those things that use the worker process infrastructure (e.g test execution).